backend/authapp/utils.py

```diff
@@ -1,12 +1,16 @@
+from utils.general import str2bool
+
 def can_modify(request, instance):
     modifying_user = request.user.username
     modified_user = instance.username
+
     modified_user_is_staff = instance.is_staff
+    modifying_user_is_staff = str2bool(request.data.get('is_staff', 'true'))
 
-    if modified_user == 'admin':
+    if modified_user == 'admin' and modifying_user_is_staff == False:
         return False, "Cannot change admin(SuperUser) account privileges"
-
-    if modifying_user == modified_user:
+    
+    if modified_user_is_staff and modifying_user == modified_user and modified_user_is_staff != modifying_user_is_staff:
```

Comment:
This conditional is not very easy to read and understand what its doing. If the user we are attempting to change is a staff member and the user we are attempting to change is ourselves and we are changing their permissions. Why is the `modified_user_is_staff` needed? Isn't it the case that nobody can change their own permissions?